test/third-party/miniflare.test.ts

```diff
@@ -0,0 +1,54 @@
+// @vitest-environment miniflare
+import { afterAll, afterEach, beforeAll, expect, test, vi } from 'vitest'
+import { BatchInterceptor } from '../../src'
+import { ClientRequestInterceptor } from '../../src/interceptors/ClientRequest'
+import { XMLHttpRequestInterceptor } from '../../src/interceptors/XMLHttpRequest'
+import { FetchInterceptor } from '../../src/interceptors/fetch'
+import { httpGet, httpsGet } from '../helpers'
+
+const interceptor = new BatchInterceptor({
+  name: 'setup-server',
+  interceptors: [
+    new ClientRequestInterceptor(),
+    new XMLHttpRequestInterceptor(),
```

Comment:
XMLHttp is not defined in miniflare, but `setupServer` in msw defaults to applying it.  it's a noop, but I've kept that here, even though it's unused - as a regression check that we can apply it

We may want to look into scoping an interceptor set for that environment?